pkg/devicetwin/dtcontext/dtcontext_test.go

```diff
@@ -0,0 +1,451 @@
+package dtcontext
+
+import (
+	"reflect"
+	"sync"
+	"testing"
+
+	"github.com/kubeedge/kubeedge/beehive/pkg/common/config"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/context"
+	"github.com/kubeedge/kubeedge/beehive/pkg/core/model"
+	"github.com/kubeedge/kubeedge/pkg/common/util"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dtcommon"
+	"github.com/kubeedge/kubeedge/pkg/devicetwin/dttype"
+)
+
+// TestInitDTContext is function to test InitDTContext().
+func TestInitDTContext(t *testing.T) {
+	util.LoadConfig()
+	nodeID, err := config.CONFIG.GetValue("edgehub.controller.node-id").ToString()
+	if err != nil {
+		t.Errorf("Error in getting node id %v", err)
+		return
+	}
+	tests := []struct {
+		name    string
+		context *context.Context
+		want    *DTContext
+		wantErr error
+	}{
+		{
+			name:    "ActualContextArgumentTest",
+			context: context.GetContext(context.MsgCtxTypeChannel),
+			want: &DTContext{
+				GroupID:        "",
+				NodeID:         nodeID,
+				CommChan:       make(map[string]chan interface{}, 1),
+				ConfirmChan:    make(chan interface{}, 1),
+				ConfirmMap:     &sync.Map{},
+				ModulesHealth:  &sync.Map{},
+				ModulesContext: context.GetContext(context.MsgCtxTypeChannel),
+				DeviceList:     &sync.Map{},
+				DeviceMutex:    &sync.Map{},
+				Mutex:          &sync.Mutex{},
+				State:          dtcommon.Disconnected,
+			},
+			wantErr: nil,
+		},
+		{
+			name:    "EmptyContextArgumentTest",
+			context: context.GetContext("test"),
+			want: &DTContext{
+				GroupID:        "",
+				NodeID:         nodeID,
+				CommChan:       make(map[string]chan interface{}),
+				ConfirmChan:    make(chan interface{}, 1),
+				ConfirmMap:     &sync.Map{},
+				ModulesHealth:  &sync.Map{},
+				ModulesContext: context.GetContext("test"),
+				DeviceList:     &sync.Map{},
+				DeviceMutex:    &sync.Map{},
+				Mutex:          &sync.Mutex{},
+				State:          dtcommon.Disconnected,
+			},
+			wantErr: nil,
+		},
+	}
+	for _, test := range tests {
+		t.Run(test.name, func(t *testing.T) {
+			got, err := InitDTContext(test.context)
+			if err != test.wantErr {
+				t.Errorf("InitDTContext() error = %v, wantErr %v", err, test.wantErr)
```

Comment:
There should be a return inside all `if` statements, if one of the value in the structure in wrong, it means that the structure received in incorrect. Sounds ok ?